File: CyFile-Android/app/src/test/java/at/tugraz/tc/cyfile/crypto/mocks/MockedKeyStore.java
package at.tugraz.tc.cyfile.crypto.mocks;

import java.security.KeyStore;
import java.security.KeyStoreSpi;
import java.security.Provider;

public class MockedKeyStore extends KeyStore {
    /**
     * Creates a KeyStore object of the given type, and encapsulates the given
     * provider implementation (SPI object) in it.
     *
     * @param keyStoreSpi the provider implementation.
     * @param provider    the provider.
     * @param type        the keystore type.
     */
    public MockedKeyStore(KeyStoreSpi keyStoreSpi, Provider provider, String type) {
        super(keyStoreSpi, provider, type);
    }
}

File: CyFile-Android/app/src/test/kotlin/at/tugraz/tc/cyfile/crypto/mocks/MockedKeyStore.kt
package at.tugraz.tc.cyfile.crypto.mocks

import java.security.KeyStore
import java.security.KeyStoreSpi
import java.security.Provider

class MockedKeyStore
/**
 * Creates a KeyStore object of the given type, and encapsulates the given
 * provider implementation (SPI object) in it.
 *
 * @param keyStoreSpi the provider implementation.
 * @param provider    the provider.
 * @param type        the keystore type.
 */
(keyStoreSpi: KeyStoreSpi, provider: Provider, type: String) : KeyStore(keyStoreSpi, provider, type)
